### PR TITLE
Add feature to sign in and sign up the user

### PR DESCRIPTION
**Pull request by bibekmaharjan**
## Description

* Added sign in and sign up pages.
* Added protected route component to verify the authentication.

### Screenshot

image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,39 @@
...
+import ProtectedRoute from './components/common/ProtectedRoute';
 
 function App() {
+  const [token, setToken] = React.useState<string | null>(localStorage.getItem('token'));
+
+  React.useEffect(() => {
+    if (token) {
+      localStorage.setItem('token', token);
+    } else {
+      localStorage.removeItem('token');
```
Comment (sbimochan) [inline src/App.tsx:19]: There can some errors while doing getItem from localStorage, can you store these actions in a localStorageUtil? The get method can have try...catch block with `void 0` in catch block

File: src/assets/styles/components/_button.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
+.btn {
```
Comment (sbimochan) [inline src/assets/styles/components/_button.scss:1]: Use 2 space indent for consistency

File: src/config.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import Config from './domain/misc/Config';
+
+const config: Config = {
+  endpoints: {
+    signIn: 'http://localhost:8080/auth/signin',
+    signUp: 'http://localhost:8080/auth/signup',
+    practitioners: 'http://localhost:8080/practitioners',
+    editPractitioners: 'http://localhost:8080/practitioners/{id}',
+    deletePractitioners: 'http://localhost:8080/practitioners/{id}',
+    practitionerDetails: 'http://localhost:8080/practitioners/{id}',
```
Comment (sbimochan) [inline src/config.ts:10]: create a separate constant for baseUrl that will retrieve value localhost from .env

File: src/domain/misc/Config.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+interface Config {
+  endpoints: {
+    signIn: string;
+    signUp: string;
+    practitioners: string;
+    editPractitioners: string;
+    deletePractitioners: string;
+    practitionerDetails: string;
```
Comment (sbimochan) [inline src/domain/misc/Config.ts:8]: you sure that these are string?

File: src/pages/Signin.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  const [password, setPassword] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const isDisabled = !email || !password || isSubmitting;
+
+  const handleSubmit = async (event: React.MouseEvent) => {
+    event.preventDefault();
+    setIsSubmitting(true);
+    try {
+      const response = await axios.post(config.endpoints.signIn, { email, password });
```
Comment (sbimochan) [inline src/pages/Signin.tsx:22]: create some httpUtils

File: src/pages/Signin.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+          className="input__text mb-md"
+          placeholder="YOUR MAIL"
+          id="email"
+          type="email"
+          value={email}
+          onChange={(event) => setEmail(event.target.value)}
+        />
+        <input
+          className="input__text mb-sm"
+          placeholder="YOUR PASSWORD"
```
Comment (sbimochan) [inline src/pages/Signin.tsx:48]: Don't use uppercase for placehoder. and MAIL can be changed to email id

**Review by sbimochan — CHANGES_REQUESTED** [review]

File: src/domain/misc/Config.ts (hunk 4 above)
Comment (bibekmaharjan) [reply src/domain/misc/Config.ts:8]: Yes, endpoint values are expected string.

**Pull request merged**